src/worker.cpp

```diff
@@ -1169,24 +1169,44 @@ types::VPDMapVariant Worker::parseVpdFile(const std::string& i_vpdFilePath)
 
     std::shared_ptr<Parser> vpdParser = std::make_shared<Parser>(i_vpdFilePath,
                                                                  m_parsedJson);
-    types::VPDMapVariant l_parsedVpd = vpdParser->parse();
-
-    // Before returning, as collection is over, check if FRU qualifies for
-    // any post action in the flow of collection.
-    // Note: Don't change the order, post action needs to be processed only
-    // after collection for FRU is successfully done.
-    if (jsonUtility::isActionRequired(m_parsedJson, i_vpdFilePath, "postAction",
-                                      "collection"))
+    try
     {
-        if (!processPostAction(i_vpdFilePath, "collection", l_parsedVpd))
+        types::VPDMapVariant l_parsedVpd = vpdParser->parse();
+
+        // Before returning, as collection is over, check if FRU qualifies for
+        // any post action in the flow of collection.
+        // Note: Don't change the order, post action needs to be processed only
+        // after collection for FRU is successfully done.
+        if (jsonUtility::isActionRequired(m_parsedJson, i_vpdFilePath,
+                                          "postAction", "collection"))
         {
-            throw std::runtime_error("Required post action failed for path " +
-                                     i_vpdFilePath +
-                                     " Aborting collection for this FRU");
+            if (!processPostAction(i_vpdFilePath, "collection", l_parsedVpd))
+            {
+                throw std::runtime_error(
+                    "Required post action failed for path " + i_vpdFilePath +
+                    " Aborting collection for this FRU");
+            }
         }
+
+        return l_parsedVpd;
     }
+    catch (std::exception& ex)
```

Comment:
l_exception ?

Reply:
We are not using `l_` for the exception object reference throughout `worker.cpp`, so to maintain consistency, I followed the same convention.

Reply:
@souvik1914581 The code of worker was written before we brought it the guidelines. Hennce, with new addition we are following that.
Please modify with l_ex.

Reply:
Done